wordpress-mobile/WordPress-Android
# Issue/13328 restore progress

Parent #13328

This PR implements the "progress" wizard step/view for the restore request and is mostly a clone of Backup Download progress with a few temporary navigations until the last step in the wizard has been implemented. 

**Merge Instructions**
- Make sure #13773 & #13775 have been merged (sorry, but they need to build upon each other)
- Remove Not Ready For Merge Label
- Merge as normal

**Cloned Classes**
RestoreProgressAdapter
RestoreProgressFragment
RestoreProgressStateListItemBuilder

**Notes**:
- Ignore styling - this is being handled separately 
- Ignore unit test placement in packages - this is being handled separately 
- RestoreActivity & RestoreViewModel (+unit tests) will continue to get updated as I move throughout the steps of the wizard

**To test:**
**Warning**: The Restore site button is hooked up, so if you press it your site will be backed up!
- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag on
- Navigate to ActivityLog
- Tap the more menu
- Select the restore option
- Tap the Restore Site button on the Details View
- Tap the Confirm restore site on the Warning View
Verify
- (1) The progress view is shown
- (2) If there is an error, the wizard will close (eventually it will go to the complete view)
- (3) When the restore finishes, the wizard will close (eventually it will go to the complete view)
- (4) Tapping the OK, notify me button, Back Button and X will close the wizard

PR submission checklist:
